check_compilation.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+# This script runs javac on all of the minimized outputs under ISSUES/ .
+# It returns 2 if any of them fail to compile, 1 if there are any malformed directories,
+# and 0 if all of them do compile.
+#
+# It is desirable that all of the expected Specimin's minimized program gets compiled, because Specimin
+# should produce independently-compilable programs.
+
+# when this script is invoked from specimin CI, 
+# compilation result will be compared with "expected" result and CI will be passed/failed accordingly.
+
+returnval=0
+compile_status_json="{"
+echo "Specimin path: $SPECIMIN"
+cd ISSUES || exit 1
+for target in * ; do
+    echo "Target = ${target}"
+    if [ "${target}" = "output.html" ] || [ "${target}" = "target_status.json" ] ||  [ "${target}" = "specimin" ] ||  [ "${target}" = "compile_status.json" ]; then 
+        continue; 
+    fi
+
+    cd "${target}/output/" || exit 1
+    # check if any directory exists inside output. If no directory there, specimin failed on the input target
+    # in that case, ignoring it.
+    directory_count="$(find ./ -mindepth 1 -type d | wc -l)"
+    if [ "$directory_count" -eq 0 ]; then 
+        echo "No directories inside ${target}/output. Ignoring it."
+        compile_status_json="$compile_status_json\n  \"$target\": \"FAIL\","
+        cd ../..
+        continue
+    fi
+
+    # javac relies on word splitting
+    # shellcheck disable=SC2046
+    javac -classpath "$SPECIMIN/main/src/test/resources/shared/checker-qual-3.42.0.jar" $(find . -name "*.java") \
+    && { echo "Running javac on ${target}/output PASSES"; compile_status_json="$compile_status_json\n  \"$target\": \"PASS\","; } \
+    || { echo "Running javac on ${target}/output FAILS. Please check logs above."; compile_status_json="$compile_status_json\n  \"$target\": \"FAIL\","; returnval=2; }
```

Comment:
This construction doesn't guarantee that the "fails" code will only run when javac is falsy. See https://www.shellcheck.net/wiki/SC2015.